contracts/implementations/AdLicenseValidator/AdLicenseValidatorInternal.sol

```diff
@@ -3,33 +3,31 @@ pragma solidity ^0.8.13;
 
 import "../../libraries/AdLicenseValidatorStorage.sol";
 
+error InvalidLicense();
+
 /// @title AdLicenseValidatorInternal
 /// @notice Contract with internal functions to assist in aftermarket device minting
 /// @dev Stake contract repository https://github.com/DIMO-Network/dimo-staking-contract-license-nft
 contract AdLicenseValidatorInternal {
-    /// @notice Validates if the manufacturer has a License
-    /// Calculates the total cost to mint the desired amount of aftermarket devices
-    /// Transfers the calculated amount to the foundation
-    /// @dev This contract must be approved to spend the tokens in advance
-    /// @param manufacturer The address of the manufacturer
-    /// @param amount The amount of devices to be minted
-    function _validateMintRequest(address manufacturer, uint256 amount)
-        internal
-    {
-        require(
-            AdLicenseValidatorStorage.getStorage().license.balanceOf(
-                manufacturer
-            ) == 1,
-            "Invalid license"
-        );
-
+    /**
+     * @notice Validates if the manufacturer has a License
+     * Calculates the total cost to mint the desired amount of aftermarket devices
+     * The sender transfers the calculated amount to the foundation
+     * @dev This contract must be approved to spend the tokens in advance
+     * @param manufacturer The address of the manufacturer
+     * @param sender The address of the sender
+     * @param amount The amount of devices to be minted
+     */
+    function _validateMintRequest(
+        address manufacturer,
+        address sender,
```

Comment:
It's interesting that this doesn't validate the permissions possessed by the `sender`.

Reply:
Well, we validate the lincense of the manufacturer and the privilege of the sender given by the manufacturer. What else do you have in mind?